src/libraries/System.Net.NetworkInformation/src/System/Net/NetworkInformation/NetworkAddressChange.Browser.cs

```diff
@@ -0,0 +1,61 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Microsoft.Win32.SafeHandles;
+
+using System.ComponentModel;
+using System.Collections.Generic;
+using System.Net.Sockets;
+using System.Runtime.Versioning;
+using System.Threading;
+
+namespace System.Net.NetworkInformation
+{
+    public partial class NetworkChange
+    {
+        private static bool s_isBrowserNetworkChangeListenerAttached;
+        private static event NetworkAvailabilityChangedEventHandler? s_networkAvailabilityChanged;
+
+        [UnsupportedOSPlatform("illumos")]
+        [UnsupportedOSPlatform("solaris")]
+        public static event NetworkAvailabilityChangedEventHandler? NetworkAvailabilityChanged
+        {
+            add
+            {
+                if (!s_isBrowserNetworkChangeListenerAttached)
+                {
+                    BrowserNetworkInterfaceInterop.AddChangeListener(OnNetworkChanged);
+                    s_isBrowserNetworkChangeListenerAttached = true;
+                }
+
+                s_networkAvailabilityChanged += value;
+            }
+            remove
+            {
+                s_networkAvailabilityChanged -= value;
+            }
+        }
+
+        private static void OnNetworkChanged(bool isOnline) => s_networkAvailabilityChanged?.Invoke(null, new NetworkAvailabilityEventArgs(isOnline));
+
+        [UnsupportedOSPlatform("browser")]
+        [UnsupportedOSPlatform("illumos")]
+        [UnsupportedOSPlatform("solaris")]
+        public static event NetworkAddressChangedEventHandler? NetworkAddressChanged
+        {
+            add => throw new System.PlatformNotSupportedException(System.SR.SystemNetNetworkInformation_PlatformNotSupported);
+            remove => throw new System.PlatformNotSupportedException(System.SR.SystemNetNetworkInformation_PlatformNotSupported);
```

Comment:
The corresponding tests should be run for browser, and check for PNSE.

Reply:
Just curious, are we writing a test for every API that is not supported on any platform?
Because this is already not supported on `illumos` and `solaris`. Should I run the test also on these platforms?